src/OrchardCore/OrchardCore.Users.Abstractions/Handlers/UpdateUserContext.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Nodes;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Users.Handlers
+{
+    /// <summary>
+    /// Represents a <see cref="UserContextBase"/> for updated user roles.
+    /// </summary>
+    public class UpdateUserContext : UserContextBase
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="UpdateUserContext"/>.
+        /// </summary>
+        /// <param name="user">The <see cref="IUser"/>.</param>
+        /// <param name="loginProvider">The login provider.</param>
+        /// <param name="externalClaims">The user claims.</param>
+        /// <param name="userProperties">The user properties.</param>
+        public UpdateUserContext(IUser user, string loginProvider, IEnumerable<SerializableClaim> externalClaims, JsonObject userProperties) : base(user)
+        {
+            ExternalClaims = externalClaims.AsEnumerable();
+            LoginProvider = loginProvider;
+            UserProperties = userProperties.DeepClone() as JsonObject;
```

Comment:
I don't think we need to clone the properties here. But if you think it is safer, keep it.